### PR TITLE
Rename `to_str(array)` and `to_array(str)` 

### LINKED ISSUE #1401: Consider renaming `to_str(array)` and `to_array(str)` to `str_split` and `array_join` respectively.
`to_str(array, delimiter)` and `to_array(str, delimiter)` are, at a glance, consistent with other conversion/constructor functions, but `to_array(str, delimiter)` is an odd and undisoverable way of naming a string-splitting function, and `str_split()` would be a better name.  On the other hand, `to_str(array, delimiter)` has a better claim at being similar to other "formatting" functions, but is also distinct in that it converts a collection type into a scalar type, so naming it `array_join` would not be too bad.

### PR DESCRIPTION
**Pull request by shankarj67**
Rename `to_str(array)` and `to_array(str)` to `str_split` and `array_join

Fixes #1401

### CODE REVIEW

File: docs/edgeql/funcops/array.rst [hunk 1]
```diff
@@ -32,10 +32,10 @@ Array
     * - :eql:func:`find`
       - Find the index of an element in the array.
 
-    * - :eql:func:`to_str`
+    * - :eql:func:`array_join`
       - Render an array to a string.
 
-    * - :eql:func:`to_array`
+    * - :eql:func:`str_split`
```
Comment (1st1) [inline docs/edgeql/funcops/array.rst:38]: This one should be moved to https://github.com/edgedb/edgedb/blob/master/docs/edgeql/funcops/string.rst

File: edb/lib/std/70-converters.edgeql [hunk 2]
```diff
@@ -238,24 +238,24 @@ std::to_str(json: std::json, fmt: OPTIONAL str={}) -> std::str
 
 
 CREATE FUNCTION
-std::to_str(array: array<std::str>, delimiter: std::str) -> std::str
+std::array_join(array: array<std::str>, delimiter: std::str) -> std::str
 {
     SET volatility := 'STABLE';
     USING SQL $$
-    SELECT array_to_string("array", "delimiter");
+    SELECT array_array_join("array", "delimiter");
```
Comment (1st1) [inline edb/lib/std/70-converters.edgeql:245]: No need for this line to be changed.

File: edb/lib/std/70-converters.edgeql [hunk 3]
```diff
@@ -238,24 +238,24 @@ std::to_str(json: std::json, fmt: OPTIONAL str={}) -> std::str
...
+std::str_split(s: std::str, delimiter: std::str) -> array<std::str>
 {
     SET volatility := 'IMMUTABLE';
     USING SQL $$
         SELECT (
             CASE WHEN "delimiter" != ''
-            THEN string_to_array("s", "delimiter")
-            ELSE regexp_split_to_array("s", '')
+            THEN string_str_split("s", "delimiter")
+            ELSE regexp_split_str_split("s", '')
```
Comment (1st1) [inline edb/lib/std/70-converters.edgeql:258]: Please revert the implementation to the old code.

File: edb/lib/std/70-converters.edgeql [hunk 4]
```diff
@@ -238,24 +238,24 @@ std::to_str(json: std::json, fmt: OPTIONAL str={}) -> std::str
...
     USING SQL $$
-    SELECT array_to_string("array", "delimiter");
+    SELECT array_array_join("array", "delimiter");
     $$;
 };
 
 
 CREATE FUNCTION
-std::to_array(s: std::str, delimiter: std::str) -> array<std::str>
+std::str_split(s: std::str, delimiter: std::str) -> array<std::str>
```
Comment (1st1) [inline edb/lib/std/70-converters.edgeql:251]: This function should be moved to https://github.com/edgedb/edgedb/blob/master/edb/lib/std/30-strfuncs.edgeql

File: edb/lib/std/70-converters.edgeql (hunk 2 above)
Comment (shankarj67) [reply edb/lib/std/70-converters.edgeql:245]: I need to change the function name only.

Comment (1st1) [reply edb/lib/std/70-converters.edgeql:245]: I'm not sure how to read your last comment, sorry.

**Comment by 1st1** [comment]
Also we need you to accept the CLA.

File: edb/lib/std/70-converters.edgeql (hunk 2 above)
Comment (shankarj67) [reply edb/lib/std/70-converters.edgeql]: I have added the new commit. Please let me know if you need any other change.

File: tests/test_edgeql_functions.py [hunk 5]
```diff
@@ -2083,7 +2083,7 @@ async def test_edgeql_functions_to_str_07(self):
                 await self.con.fetchall(
                     r'''SELECT to_str(<cal::local_time>'15:01:22', '');''',)
 
-    async def test_edgeql_functions_to_str_08(self):
+    async def test_edgeql_functions_array_join_01(self):
         await self.assert_query_result(
             r'''SELECT to_str(['one', 'two', 'three'], ', ');''',
```
Comment (1st1) [inline tests/test_edgeql_functions.py:2088]: `SELECT to_str(['one', 'two', 'three'], ', ');` needs to be changed to `SELECT array_join(['one', 'two', 'three'], ', ');`

File: tests/test_edgeql_functions.py [hunk 6]
```diff
@@ -2099,7 +2099,7 @@ async def test_edgeql_functions_to_str_08(self):
             [''],
         )
 
-    async def test_edgeql_functions_to_array_01(self):
+    async def test_edgeql_functions_str_split_01(self):
         await self.assert_query_result(
             r'''SELECT to_array('one, two, three', ', ');''',
```
Comment (1st1) [inline tests/test_edgeql_functions.py:2104]: to_array -> str_split

File: edb/lib/std/70-converters.edgeql (hunk 2 above)
Comment (1st1) [reply edb/lib/std/70-converters.edgeql]: > I have added the new commit. Please let me know if you need any other change.

Ideally you should set up local development environment (https://edgedb.com/docs/internals/dev) and run the test suite. The PR has some errors that would make our test suite unhappy.

File: tests/test_edgeql_functions.py (hunk 5 above)
Comment (shankarj67) [reply tests/test_edgeql_functions.py:2088]: Sorry. I did not check this. I will do it for both functions.

File: edb/lib/std/70-converters.edgeql (hunk 2 above)
Comment (shankarj67) [reply edb/lib/std/70-converters.edgeql]: Yes. I am running the test suite locally. I will add the commit once it is done.

Comment (shankarj67) [reply edb/lib/std/70-converters.edgeql]: I have changed everything. But I am getting an SSLerror while setting up edgedb on my laptop. I am unable to run the test suite. Shall I send the new commit?

Comment (1st1) [reply edb/lib/std/70-converters.edgeql]: Well, you need to update references to the functions you renamed:

1. sql.edgeql:178: to_str -> str_join
2. repl/__init__.py:689,760 to_str -> str_join

**Comment by vpetrovykh** [comment]
Closing in favor of #1415

**Pull request closed without merging**